components/match2/commons/match_group_base.lua

```diff
@@ -6,23 +6,20 @@
 -- Please see https://github.com/Liquipedia/Lua-Modules to contribute
 --
 
+local Array = require('Module:Array')
 local FeatureFlag = require('Module:FeatureFlag')
+local Json = require('Module:Json')
 local Logic = require('Module:Logic')
 local Lua = require('Module:Lua')
 local Match = require('Module:Match')
-local String = require('Module:StringUtils')
 local Variables = require('Module:Variables')
-local json = require('Module:Json')
 
 local MatchGroupDisplay = Lua.import('Module:MatchGroup/Display', {requireDevIfEnabled = true})
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
 
 local category = ''
 local _loggedInWarning = ''
 
-local PARENT = Variables.varDefault('tournament_parent', '')
-
-local BRACKET_DATA_PARAMS = {'header', 'tolower', 'toupper', 'qualwin', 'quallose', 'skipround'}
-
 local p = {}
```

Comment:
while at it you could rename "p" to "matchGroup" or something more descriptive ;)